### DISCUSSION TITLE
Skipped Resources

### DISCUSSION THREAD

**Discussion by paulmccrady** [Azure/PSRule.Rules.Azure#2480] [Q&A] [answered]
I've been trialling running the PSRules.Rules.Azure rule set against some Azure resources and have so far been very impressed. I was wondering if it is possible to get a list of resources that were not included in a set of test results? That way I know a delta of resources included within the rule set and those not.

**Comment by BernieWhite** [comment, marked as answer]
@paulmccrady You want to know if PSRule for Azure didn't run any rules against some resources/ resource types, because these would be a gap in coverage until you or the PSRule for Azure project build these rules.

Did I understand the question correctly?

---

Today, PSRule generates a warning on any resources where no rules match/ are run.

However, customers often disable this warning by the following to reduce output noise:

- Setting `execution.notProcessedWarning` to `false` - This way is depreciated and will be removed in PSRule v3.
- Setting `execution.unprocessedObject` to `Ignore` - This is supported in PSRule v2.9.0 or greater.

https://microsoft.github.io/PSRule/v2/concepts/PSRule/en-US/about_PSRule_Options/#executionnotprocessedwarning
https://microsoft.github.io/PSRule/v2/concepts/PSRule/en-US/about_PSRule_Options/#executionunprocessedobject

We are working to improve output noise so there is fewer cases where this option is disabled.

---

In summary I think that by re-enabling this warning it would meet your requirements. If not, how would you like it to work?

**Reply by paulmccrady** [reply]
@BernieWhite 
Did I understand the question correctly? - yes
If not, how would you like it to work? - Report the resources as skipped (or some other similar alternative) in test results. Checking the console output would be more effort.
